opba-rest-api-shared/src/main/java/de/adorsys/opba/restapi/shared/service/FacadeResponseMapper.java

```diff
@@ -93,4 +102,33 @@
         result.getHeaders().forEach(builder::header);
         return builder;
     }
+
+    private ResponseEntity.BodyBuilder setCookieHeaders(FacadeResultRedirectable<?, ?> result, ResponseEntity.BodyBuilder builder) {
+        if (StringUtils.isBlank(result.getToken())) {
+            return builder;
+        }
+
+        String ttl = Long.toString(cookieProperties.getRedirectMaxAge().getSeconds());
+        String cookie = redirectCookieBuilder(result.getAuthorizationSessionId(), result.getRedirectCode(), AUTHORIZATION_SESSION_KEY, result.getToken())
+                                .build().toString();
+
+        builder.header(COOKIE_TTL, ttl)
+                .header(SET_COOKIE, cookie);
+
+        return builder;
+    }
+
+    public ResponseCookie.ResponseCookieBuilder redirectCookieBuilder(String authorizationId, String redirectState, String name, String value) {
```

Comment:
To avoid cookie building logic duplication with

```
 private String[] buildAuthorizationCookiesOnAllPaths(UUID authorizationId, String key) {
        String token = authService.generateToken(key);
        return authConfig.getAuthorizationSessionKey().getCookie().getPathTemplates().stream()
                .map(it -> cookieString(authorizationId, it, token))
                .toArray(String[]::new);
    }

    private String cookieString(UUID authorizationId, String path, String token) {
        ResponseCookie.ResponseCookieBuilder builder = tppAuthResponseCookieBuilder
                .builder(AUTHORIZATION_SESSION_KEY,  token);

        if (!Strings.isNullOrEmpty(authConfig.getAuthorizationSessionKey().getCookie().getDomain())) {
            builder = builder.domain(authConfig.getAuthorizationSessionKey().getCookie().getDomain());
        }

        builder = builder.path(
                UriComponentsBuilder.fromPath(path)
                        .buildAndExpand(ImmutableMap.of(UriExpandConst.AUTHORIZATION_SESSION_ID, authorizationId.toString()))
                        .toUriString()
        );

        return builder.build().toString();
    }
```

I suggest creating CookieBuilder class in the facade that will be used by security-api-impl and rest-impl

Reply:
done